src/share/ash/ob_active_sess_hist_task.cpp

```diff
@@ -180,6 +183,14 @@ void ObActiveSessHistTask::runTimerTask()
         }
       }
     } // end for
+    elapsed_secs_++;
+    if (elapsed_secs_ % snapshot_interval == 0) {
+      elapsed_secs_ = 0;
+      if (do_snapshot_ahead()) {
+        // send rpc
```

Comment:
should we update read pos after send rpc?

Reply:
this pr is only for decision-making for snapshot ahead, internal state change will be handled in issue #2026 and #2027

Reply:
TODO: read_pos_ modification
1. snapshot ahead
2. rpc broadcast
3. resize_ash_size